agent/async/ttl_cache.go

```diff
@@ -0,0 +1,79 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package async
+
+import (
+	"sync"
+	"time"
+)
+
+type TTLCache interface {
+	// Get fetches a value from cache, returns nil, false on miss
+	Get(key string) (value interface{}, expired bool, ok bool)
+	// Set sets a value in cache. overrites any existing value
+	Set(key string, value interface{})
+	// Delete deletes the value from the cache
+	Delete(key string)
+}
+
+// Creates a TTL cache with ttl for items.
+func NewTTLCache(ttl time.Duration) TTLCache {
+	return &ttlCache{
+		ttl:      ttl,
+		cache:    make(map[string]interface{}),
+		expiries: make(map[string]time.Time),
+	}
+}
+
+type ttlCache struct {
```

Comment:
I think it would be better to have a cache entry and avoid having two maps:
```golang
type ttlCacheEntry struct {
    value interface{}
    expiry time.Time
}

type ttlCache struct {
    cache map[string]ttlCacheEntry
    . 
    .
    .
}
```